Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_preferences.py

```diff
@@ -167,33 +170,62 @@ def update_project_info(self):
 
     def on_project_path_updated(self, context):
         self.update_project_info()
-
+        self.join_create_project_thread()
 
     def monitor_package_lock(self, context):
-        while not self.cancel_thread:
+        while not self.cancel_package_lock_thread:
             sleep(0.5)
             if msb_meshsync_version_package_lock(self.project_path) != "":
                 self.is_meshsync_in_manifest_lock = True
                 self.redraw(context)
                 return
 
-
     def on_in_package_lock_updated(self, context):
-        if self.thread is not None:
-            self.cancel_thread = True
-            self.thead.join(2.0)
+        if self.package_lock_thread is not None:
+            self.cancel_package_lock_thread = True
+            self.package_lock_thread.join(2.0)
 
         if not self.is_meshsync_in_manifest_lock:
-            self.thread = Thread(target = self.monitor_package_lock, args = (context,), daemon = True)
-            self.cancel_thread = False
-            self.thread.start()
-            atexit.unregister(self.shutdown_thread)
-            atexit.register(self.shutdown_thread)
+            self.package_lock_thread = Thread(target = self.monitor_package_lock, args = (context,), daemon = True)
+            self.cancel_package_lock_thread = False
+            self.package_lock_thread.start()
+            atexit.unregister(self.shutdown_threads)
+            atexit.register(self.shutdown_threads)
+
+    def join_create_project_thread(self):
+        if self.create_project_thread is not None:
+            self.cancel_create_project_thread = True
+            self.thead.join(2.0)
 
-    def shutdown_thread():
-        MESHSYNC_Preferences.cancel_thread = True
-        if MESHSYNC_Preferences.thread is not None:
-            MESHSYNC_Preferences.thread.join(2.0)
+    def monitor_project_created(self, context):
+        while not self.cancel_create_project_thread:
+            sleep(0.5)
+            if msb_validate_project_path(self.project_path):
+                self.is_project_created = False
+                self.is_unity_project = True
+                self.redraw(context)
+                return
```

Comment:
Does this need a timeout so it doesn't get caught forever if the project path check keeps failing?

Reply:
I would be against that. The reason is we cannot have an accurate and meaningful timeout as each project takes minutes to create, and that can vary depending on the template, os resources, hardware etc.  This thread does not block the main thread and it will be terminated before a new thread with the same job is created. Also, Unity failing to create a project is a very unlikely scenario.